### PR TITLE
Update sync space to new api

### PR DESCRIPTION
**Pull request by mhpjay422**
Why this PR?
- Update sync space to new api

### CODE REVIEW

File: lib/flatfile.ts [hunk 1]
```diff
@@ -248,3 +252,85 @@ export const postFile = async ({
...
+  });
+
+  if (!workbookObj) {
+    throw "Could not find workbook";
+  }
+
+  const sheets = workbookObj.sheets?.find((s) => s.name === sheetName);
+
+  if (!sheets) {
+    throw "Could not find sheets";
```
Comment (johnmosesman) [inline lib/flatfile.ts:329]: What catches this further up the chain? Should we try/catch here and return a reasonable error + not throw?

File: lib/sync-records.ts [hunk 2]
```diff
@@ -51,129 +49,127 @@ export const syncWorkbookRecords = async ({
...
-      let recordJobCode = () => {
-        if (values.jobCode && values.jobCode.value) {
-          return values.jobCode.value as string;
-        } else {
-          let jobName = values.jobName.value as string;
-
-          return jobName.replaceAll(" ", "_");
-        }
-      };
+  if (validsManagersFirst) {
```
Comment (johnmosesman) [inline lib/sync-records.ts:66]: Why's this in an if? Shouldn't necessarily be required, and where's the else case?

File: lib/sync-records.ts [hunk 3]
```diff
@@ -1,48 +1,46 @@
...
     workbookName: process.env.ONBOARDING_WORKBOOK_NAME as string,
     sheetName: "Jobs",
     spaceType,
   });
 
-  const totalRecords = employeeRecords.length + jobRecords.length;
+  const numEmployeeRecords = employeeRecords?.length
+    ? employeeRecords.length
+    : 0;
+  const numJobRecords = jobRecords?.length ? jobRecords.length : 0;
```
Comment (johnmosesman) [inline lib/sync-records.ts:41]: What's the return type here? Is it an empty array if it succeeds? Maybe a `jobRecords?.length ?? 0` or something would work

File: lib/flatfile.ts (hunk 1 above)
Comment (mhpjay422) [reply lib/flatfile.ts:329]: getRecords needs the sheetId. 

Wrapping the whole function in a try catch

File: lib/sync-records.ts (hunk 3 above)
Comment (mhpjay422) [reply lib/sync-records.ts:41]: ?? is much cleaner here to get the number of records

File: lib/sync-records.ts (hunk 2 above)
Comment (mhpjay422) [reply lib/sync-records.ts:66]: setting a console log here in case there are no employees to upsert

File: lib/sync-records.ts [hunk 4]
```diff
@@ -51,129 +47,129 @@ export const syncWorkbookRecords = async ({
...
-
-  console.log("Valid job records to sync", validJobs.length);
-
-  const upsertJobs = await upsertJobRecords(validJobs, {
-    userId,
-    organizationId,
-  });
-
-  const validEmployees = await validEmployeeRecords(employeeRecords);
+  console.log("Valid job records to sync", numJobRecords);
```
Comment (johnmosesman) [inline lib/sync-records.ts:50]: let's start commenting out or removing console logs

File: lib/flatfile.ts [hunk 5]
```diff
@@ -248,3 +252,90 @@ export const postFile = async ({
...
+  workbookName: string;
+  sheetName: string;
+}) => {
+  try {
+    const response = await listWorkbooks({
+      workflow,
+      spaceId: flatfileSpaceId,
+    });
+
+    console.log("getWorkbooks response", response);
```
Comment (johnmosesman) [inline lib/flatfile.ts:323]: let's start commenting out or removing console logs

File: lib/sync-records.ts [hunk 6]
```diff
@@ -51,129 +47,129 @@ export const syncWorkbookRecords = async ({
...
-      let recordJobCode = () => {
-        if (values.jobCode && values.jobCode.value) {
-          return values.jobCode.value as string;
-        } else {
-          let jobName = values.jobName.value as string;
-
-          return jobName.replaceAll(" ", "_");
-        }
-      };
+  if (!validsManagersFirst) {
```
Comment (johnmosesman) [inline lib/sync-records.ts:64]: What's the return type of employeeRecords, if it's an empty array we don't need this if check. Does this error without this if? An if with just a log isn't really worth it

File: lib/sync-records.ts (hunk 4 above)
Comment (mhpjay422) [reply lib/sync-records.ts:50]: Commented out

File: lib/flatfile.ts (hunk 5 above)
Comment (mhpjay422) [reply lib/flatfile.ts:323]: Commented out

File: lib/sync-records.ts (hunk 6 above)
Comment (mhpjay422) [reply lib/sync-records.ts:64]: RecordsWithLinks | null | undefined

The await Promise.all(upsertEmployees) wont accept upsertEmployees as possibly undefined. 

Putting the await Promise.all(upsertEmployees) in a conditional?


The error im getting is a overload error on employeeRecords. Getting this TS error

```
No overload matches this call.
  Overload 1 of 2, '(values: readonly unknown[] | []): Promise<unknown[] | []>', gave the following error.
    Argument of type 'Promise<void>[] | undefined' is not assignable to parameter of type 'readonly unknown[] | []'.
      Type 'undefined' is not assignable to type 'readonly unknown[] | []'.
  Overload 2 of 2, '(values: Iterable<void | PromiseLike<void>>): Promise<void[]>', gave the following error.
    Argument of type 'Promise<void>[] | undefined' is not assignable to parameter of type 'Iterable<void | PromiseLike<void>>'.
      Type 'undefined' is not assignable to type 'Iterable<void | PromiseLike<void>>'.ts(2769)
```

**Comment by johnmosesman** [comment]
merged in other pr

**Pull request closed without merging**